modules/core/04-channel/types/upgrade.go

```diff
@@ -69,3 +71,22 @@ func (uf UpgradeFields) ValidateBasic() error {
 func (ut UpgradeTimeout) IsValid() bool {
 	return !ut.Height.IsZero() || ut.Timestamp != 0
 }
+
+// HasPassed returns true if the upgrade has passed the timeout height or timestamp
+func (ut UpgradeTimeout) HasPassed(ctx sdk.Context) (bool, error) {
+	if !ut.IsValid() {
+		return true, errorsmod.Wrap(ErrInvalidUpgrade, "upgrade timeout cannot be empty")
```

Comment:
dropping the comment from #3536 here

shouldn't this be `ErrInvalidUpgradeTimeout`? To mirror its usage in ValidateBasic of MsgChannelUpgradeTry:

```suggestion
		return true, errorsmod.Wrap(ErrInvalidUpgradeTimeout, "timeout height or timeout timestamp must be non-zero")
```